File: msa-spring/src/test/java/com/chenluo/controller/App1Test.java
package com.chenluo.controller;

import com.chenluo.MysqlTestBase;
import com.chenluo.service.StatefulService;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

@SpringBootTest()
public class App1Test extends MysqlTestBase {
    private final StatefulService service;

    @Autowired
    public App1Test(StatefulService service) {
        this.service = service;
    }

    @Test
    public void test() {
        service.increase();
    }

    @Test
    public void get() {
        System.out.println(service.get());
    }
}

File: msa-spring-kotlin/src/test/kotlin/com/chenluo/test/kotlin/App1Test.kt
package com.chenluo.test.kotlin

import com.chenluo.test.kotlin.service.MyService
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest
internal class App1Test: BaseTest {
  @Autowired private lateinit var myService: MyService


  @Test
  fun init() {
    println(this.javaClass.name)
  }
}
